interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -198,6 +197,24 @@ private[spel] class Typer(
         case d: TypedDict                    => dictTyper.typeDictValue(d, e).map(toNodeResult)
         case TypedUnion(possibleTypes)       => typeUnion(e, possibleTypes)
         case TypedTaggedValue(underlying, _) => typeIndexer(e, underlying)
+        case TypedObjectTypingResult(fields, _, _) =>
+          def stringValue(s: StringLiteral): String = {
+            s.toString.substring(1, s.toString.length - 1)
+          }
+          val indexKeyOpt = e.childrenHead match {
+            case impl: StringLiteral => Some(stringValue(impl))
+            case _                   => None
+          }
+          val fieldOpt = for {
+            indexKey <- indexKeyOpt
+            field    <- fields.find(_._1 == indexKey)
+          } yield field
+
+          val result = fieldOpt.map(f => validNodeResult(f._2)).getOrElse(validNodeResult(Unknown))
```

Comment:
For defined `fieldIndexedByLiteralStringOpt` we should always return type. `dynamicPropertyAccessAllowed` should be checked only when is used other form of indexing, e.g. `#record[someDynamicVariable]`

Reply:
It is not done correctly currently. You catch in this case things that should be handled by dynamic property access check case

Reply:
In new approach this should be ok. Error is thrown if index key cannot be evaluated at compile time or the field in the indexed record is not present at compile time.

Reply:
I'll resolve this comment and open the new one to make it more clear what I'm talking about :)